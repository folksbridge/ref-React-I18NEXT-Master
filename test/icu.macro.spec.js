import pluginTester from 'babel-plugin-tester';
import plugin from 'babel-plugin-macros';

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: { filename: __filename, parserOpts: { plugins: ['jsx'] } },
  tests: [
    `
      import { Trans } from '../icu.macro'

      const x = <Trans>Welcome, { name }!</Trans>
    `,

    `
      import { Trans } from '../icu.macro'

      const x = <Trans>Welcome, <strong>{ name }</strong>!</Trans>
    `,

    `
      import { Trans } from '../icu.macro'

      const x = <Trans>Trainers: { trainersCount, number }</Trans>
    `,

    `
      import { Trans } from '../icu.macro'

      const x = <Trans>Trainers: <strong>{ trainersCount, number }</strong>!</Trans>
    `,

    `
      import { Trans } from '../icu.macro'

      const x = <Trans>Caught on { catchDate, date, short }</Trans>
    `,

    `
      import { Trans } from '../icu.macro'

      const x = <Trans>Caught on <strong>{ catchDate, date, short }</strong>!</Trans>
    `,

    `
      import { Select } from '../icu.macro'

      const x = <Select
        switch={gender}
        male="He avoids bugs."
        female="She avoids bugs."
        other="They avoid bugs."
      />
    `,

    `
      import { Select } from '../icu.macro'

      const x = <Select
        switch={gender}
        male={<Trans><strong>He</strong> avoids bugs.</Trans>}
        female={<Trans><strong>She</strong> avoids bugs.</Trans>}
        other={<Trans><strong>They</strong> avoid bugs.</Trans>}
      />
    `,

    `
      import { Plural } from '../icu.macro'

      const x = <Plural
        count={itemsCount1}
        $0="There is no item."
        one="There is # item."
        other="There are # items."
      />
    `,

    `
      import { Plural } from '../icu.macro'

      const x = <Plural
        i18nKey="testKey"
        count={itemsCount3}
        $0={<Trans>There is <strong>no</strong> item.</Trans>}
        one={<Trans>There is <strong>#</strong> item.</Trans>}
        other={<Trans>There are <strong>#</strong> items.</Trans>}
      />
    `,
  ],
});
