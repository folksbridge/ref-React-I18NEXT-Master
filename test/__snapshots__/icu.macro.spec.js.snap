// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1. macros 1`] = `
"
import { Trans } from '../icu.macro'

const x = <Trans>Welcome, { name }!</Trans>

      ↓ ↓ ↓ ↓ ↓ ↓

import { Trans } from 'react/i18next';


const x = <Trans defaults=\\"Welcome, {name}!\\" components={[]} values={{
  name
}} />;
"
`;

exports[`2. macros 1`] = `
"
import { Trans } from '../icu.macro'

const x = <Trans>Welcome, <strong>{ name }</strong>!</Trans>

      ↓ ↓ ↓ ↓ ↓ ↓

import { Trans } from 'react/i18next';


const x = <Trans defaults=\\"Welcome, <0>{name}</0>!\\" components={[<strong>{name}</strong>]} values={{
  name
}} />;
"
`;

exports[`3. macros 1`] = `
"
import { Trans } from '../icu.macro'

const x = <Trans>Trainers: { trainersCount, number }</Trans>

      ↓ ↓ ↓ ↓ ↓ ↓

import { Trans } from 'react/i18next';


const x = <Trans defaults=\\"Trainers: {trainersCount, number}\\" components={[]} values={{
  trainersCount
}} />;
"
`;

exports[`4. macros 1`] = `
"
import { Trans } from '../icu.macro'

const x = <Trans>Trainers: <strong>{ trainersCount, number }</strong>!</Trans>

      ↓ ↓ ↓ ↓ ↓ ↓

import { Trans } from 'react/i18next';


const x = <Trans defaults=\\"Trainers: <0>{trainersCount, number}</0>!\\" components={[<strong>{(trainersCount)}</strong>]} values={{
  trainersCount
}} />;
"
`;

exports[`5. macros 1`] = `
"
import { Trans } from '../icu.macro'

const x = <Trans>Caught on { catchDate, date, short }</Trans>

      ↓ ↓ ↓ ↓ ↓ ↓

import { Trans } from 'react/i18next';


const x = <Trans defaults=\\"Caught on {catchDate, date, short}\\" components={[]} values={{
  catchDate
}} />;
"
`;

exports[`6. macros 1`] = `
"
import { Trans } from '../icu.macro'

const x = <Trans>Caught on <strong>{ catchDate, date, short }</strong>!</Trans>

      ↓ ↓ ↓ ↓ ↓ ↓

import { Trans } from 'react/i18next';


const x = <Trans defaults=\\"Caught on <0>{catchDate, date, short}</0>!\\" components={[<strong>{(catchDate)}</strong>]} values={{
  catchDate
}} />;
"
`;

exports[`7. macros 1`] = `
"
import { Select } from '../icu.macro'

const x = <Select
  switch={gender}
  male=\\"He avoids bugs.\\"
  female=\\"She avoids bugs.\\"
  other=\\"They avoid bugs.\\"
/>

      ↓ ↓ ↓ ↓ ↓ ↓

import { Trans } from \\"react/i18next\\";


const x = <Trans defaults=\\"{gender, select,  male {He avoids bugs.} female {She avoids bugs.} other {They avoid bugs.}}\\" components={[]} values={{
  gender
}} />;
"
`;

exports[`8. macros 1`] = `
"
import { Select } from '../icu.macro'

const x = <Select
  switch={gender}
  male={<Trans><strong>He</strong> avoids bugs.</Trans>}
  female={<Trans><strong>She</strong> avoids bugs.</Trans>}
  other={<Trans><strong>They</strong> avoid bugs.</Trans>}
/>

      ↓ ↓ ↓ ↓ ↓ ↓

import { Trans } from 'react/i18next';


const x = <Trans defaults=\\"{gender, select,  male {<0>He</0> avoids bugs.} female {<1>She</1> avoids bugs.} other {<2>They</2> avoid bugs.}}\\" components={[<strong>He</strong>, <strong>She</strong>, <strong>They</strong>]} values={{
  gender
}} />;
"
`;

exports[`9. macros 1`] = `
"
import { Plural } from '../icu.macro'

const x = <Plural
  count={itemsCount1}
  $0=\\"There is no item.\\"
  one=\\"There is # item.\\"
  other=\\"There are # items.\\"
/>

      ↓ ↓ ↓ ↓ ↓ ↓

import { Trans } from \\"react/i18next\\";


const x = <Trans defaults=\\"{itemsCount1, plural,  =0 {There is no item.} one {There is # item.} other {There are # items.}}\\" components={[]} values={{
  itemsCount1
}} />;
"
`;

exports[`10. macros 1`] = `
"
import { Plural } from '../icu.macro'

const x = <Plural
  i18nKey=\\"testKey\\"
  count={itemsCount3}
  $0={<Trans>There is <strong>no</strong> item.</Trans>}
  one={<Trans>There is <strong>#</strong> item.</Trans>}
  other={<Trans>There are <strong>#</strong> items.</Trans>}
/>

      ↓ ↓ ↓ ↓ ↓ ↓

import { Trans } from \\"react/i18next\\";


const x = <Trans i18nKey=\\"testKey\\" defaults=\\"{itemsCount3, plural,  =0 {There is <0>no</0> item.} one {There is <1>#</1> item.} other {There are <2>#</2> items.}}\\" components={[<strong>no</strong>, <strong>#</strong>, <strong>#</strong>]} values={{
  itemsCount3
}} />;
"
`;
